salt/states/win_system.py

```diff
@@ -167,6 +167,54 @@ def hostname(name):
     return ret
 
 
+def workgroup(name):
+    '''
+    .. versionadded:: 2019.2.0
+
+    Manage the workgroup of the computer
+
+    name
+        The workgroup to set
+
+    Example:
+
+    .. code-block:: yaml
+
+        set workgroup:
+          system.workgroup:
+            - name: local
+    '''
+    ret = {
+        'name': name.upper(),
+        'changes': {},
+        'result': True,
+        'comment': ''
+    }
+
+    res = __salt__['system.get_domain_workgroup']()
+    current_workgroup = res['Domain'] if 'Domain' in res else res['Workgroup'] if 'Workgroup' in res else ''
+
+    if __opts__['test']:
+        ret['result'] = None
+        ret['comment'] = 'Computer will be joined to workgroup \'{0}\''.format(name)
+        return ret
+
+    if current_workgroup.upper() == name.upper():
+        ret['comment'] = "Workgroup is already set to '{0}'".format(name.upper())
+        return ret
+
+    out = __salt__['system.set_domain_workgroup'](name.upper())
+
+    if out:
```

Comment:
In order to idempotent, the state needs to check the actual value to make it has changed instead of checking for a successful run of the command.

Reply:
See step 2 and step 6:
https://docs.saltstack.com/en/latest/ref/states/writing.html#best-practices

Reply:
K, got it. Figured the initial comparison would result in the termination of the function before it even got to that point.